src/utils/c_utils/src/string_c_utils.c

```diff
@@ -440,13 +442,17 @@ bool ADUC_Safe_StrCopyN(char* dest, size_t destByteLen, const char* src, size_t
         return false;
     }
 
-    if (srcByteLen > ADUC_StrNLen(src, srcByteLen))
+    for (srcIndex = 0; (srcIndex < srcByteLen) && (src[srcIndex] != '\0'); ++srcIndex)
     {
-        return false;
+        *dest++ = src[srcIndex];
     }
+    *dest = '\0';
 
-    memcpy(dest, src, srcByteLen);
-    dest[srcByteLen] = '\0';
+    if ((srcIndex < srcByteLen) && (src[srcIndex] == '\0'))
```

Comment:
see my comment in another PR you have out.

Reply:
I'm inclined to "won't fix" this due to the following:
- The purpose is to allow caller to specify a substring or num bytes of a non-null-terminated source string.
- Caller knows the character encoding so can translate to number of bytes for the encoded characters.
- Also, can't assume the src string is null-terminated but this'll make sure the dest buffer is.
- Here is the UT that exercises this branch:
https://github.com/Azure/iot-hub-device-update/blob/ca5643f81c66905429c110b9d8da7e81f491feb4/src/utils/c_utils/tests/c_utils_ut.cpp#L507